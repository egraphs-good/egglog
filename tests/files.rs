use egg_smol::*;

fn test_program(program: &str, message: &str) {
    let mut egraph = EGraph::default();
    match egraph.parse_and_run_program(program) {
        Ok(msgs) => {
            for msg in msgs {
                log::info!("  {}", msg);
            }
        }
        Err(err) => panic!("{}: {err}", message),
    }
}

fn run(path: &str) {
    let _ = env_logger::builder().is_test(true).try_init();
    let program = std::fs::read_to_string(path).unwrap();
    test_program(&program, "Top level error");

    let egraph = EGraph::default();
    let program_str = egraph
        .parse_program(&program)
        .unwrap()
        .into_iter()
        .map(|x| x.to_string())
        .collect::<Vec<String>>()
        .join("\n");
    test_program(
        &program_str,
        &format!(
            "Program:\n{}\n ERROR after parse, to_string, and parse again.",
            program_str
        ),
    );
}

// include the tests generated from the build script
include!(concat!(std::env!("OUT_DIR"), "/files.rs"));
