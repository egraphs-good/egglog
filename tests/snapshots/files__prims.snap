---
source: tests/files.rs
expression: serialized
---
nodes:
  i64-5535299385761720939:
    op: "99999999"
    children: []
    eclass: i64-5535299385761720939
    cost: 0
  infinity-0:
    op: "infinity"
    children: []
    eclass: i64-5535299385761720939
    cost: 9223372036854776000
  Unit-0:
    op: ()
    children: []
    eclass: Unit-0
    cost: 0
  true-0:
    op: "true"
    children: []
    eclass: Unit-0
    cost: 1
  i64-5871781006564002453:
    op: "1"
    children: []
    eclass: i64-5871781006564002453
    cost: 0
  vertex-included-5871781006564002453:
    op: vertex-included
    children:
      - i64-5871781006564002453
    eclass: i64-5871781006564002453
    cost: 1
  i64-5040379952546458196:
    op: "4"
    children: []
    eclass: i64-5040379952546458196
    cost: 0
  vertex-included-5040379952546458196:
    op: vertex-included
    children:
      - i64-5040379952546458196
    eclass: i64-5871781006564002453
    cost: 1
  i64-11743562013128004906:
    op: "2"
    children: []
    eclass: i64-11743562013128004906
    cost: 0
  vertex-included-11743562013128004906:
    op: vertex-included
    children:
      - i64-11743562013128004906
    eclass: i64-5871781006564002453
    cost: 1
  i64-17615343019692007359:
    op: "3"
    children: []
    eclass: i64-17615343019692007359
    cost: 0
  vertex-included-17615343019692007359:
    op: vertex-included
    children:
      - i64-17615343019692007359
    eclass: i64-5871781006564002453
    cost: 1
  current-iteration-0:
    op: current-iteration
    children: []
    eclass: i64-17615343019692007359
    cost: 1
  i64-0:
    op: "0"
    children: []
    eclass: i64-0
    cost: 0
  iteration-to-best-edge-0:
    op: iteration-to-best-edge
    children:
      - i64-0
    eclass: "1"
    cost: 1
  iteration-to-best-edge-5871781006564002453:
    op: iteration-to-best-edge
    children:
      - i64-5871781006564002453
    eclass: "0"
    cost: 1
  iteration-to-best-edge-11743562013128004906:
    op: iteration-to-best-edge
    children:
      - i64-11743562013128004906
    eclass: "3"
    cost: 1
  solution-3877830492085173579:
    op: solution
    children:
      - i64-17615343019692007359
      - i64-5040379952546458196
      - i64-17615343019692007359
    eclass: Unit-0
    cost: 1
  solution-4591408876226341927:
    op: solution
    children:
      - i64-5871781006564002453
      - i64-5040379952546458196
      - i64-5871781006564002453
    eclass: Unit-0
    cost: 1
  solution-5712790014045719483:
    op: solution
    children:
      - i64-5871781006564002453
      - i64-11743562013128004906
      - i64-11743562013128004906
    eclass: Unit-0
    cost: 1
  iteration-to-best-edge-weight-0:
    op: iteration-to-best-edge-weight
    children:
      - i64-0
    eclass: i64-5871781006564002453
    cost: 1
  iteration-to-best-edge-weight-5871781006564002453:
    op: iteration-to-best-edge-weight
    children:
      - i64-5871781006564002453
    eclass: i64-11743562013128004906
    cost: 1
  iteration-to-best-edge-weight-11743562013128004906:
    op: iteration-to-best-edge-weight
    children:
      - i64-11743562013128004906
    eclass: i64-17615343019692007359
    cost: 1
  iteration-to-best-edge-weight-17615343019692007359:
    op: iteration-to-best-edge-weight
    children:
      - i64-17615343019692007359
    eclass: i64-5535299385761720939
    cost: 1
  edge-in-mst-5871781006564002453:
    op: edge-in-mst
    children:
      - Edge-12184509778494182535
    eclass: Unit-0
    cost: 1
  edge-in-mst-0:
    op: edge-in-mst
    children:
      - Edge-3476130915693767771
    eclass: Unit-0
    cost: 1
  edge-in-mst-17615343019692007359:
    op: edge-in-mst
    children:
      - Edge-14928484768540309164
    eclass: Unit-0
    cost: 1
  edge-exists-0:
    op: edge-exists
    children:
      - iteration-to-best-edge-5871781006564002453
    eclass: Unit-0
    cost: 1
  edge-exists-5871781006564002453:
    op: edge-exists
    children:
      - iteration-to-best-edge-0
    eclass: Unit-0
    cost: 1
  edge-exists-11743562013128004906:
    op: edge-exists
    children:
      - Edge-6437614685926927011
    eclass: Unit-0
    cost: 1
  edge-exists-17615343019692007359:
    op: edge-exists
    children:
      - iteration-to-best-edge-11743562013128004906
    eclass: Unit-0
    cost: 1
  Edge-5712790014045719483:
    op: Edge
    children:
      - i64-5871781006564002453
      - i64-11743562013128004906
      - i64-11743562013128004906
    eclass: "0"
    cost: 1
  Edge-4591408876226341927:
    op: Edge
    children:
      - i64-5871781006564002453
      - i64-5040379952546458196
      - i64-5871781006564002453
    eclass: "1"
    cost: 1
  Edge-13088382136126527480:
    op: Edge
    children:
      - i64-11743562013128004906
      - i64-5040379952546458196
      - i64-11743562013128004906
    eclass: "2"
    cost: 1
  Edge-3877830492085173579:
    op: Edge
    children:
      - i64-17615343019692007359
      - i64-5040379952546458196
      - i64-17615343019692007359
    eclass: "3"
    cost: 1
  Edge-6437614685926927011:
    op: Edge
    children:
      - i64-5040379952546458196
      - i64-11743562013128004906
      - i64-11743562013128004906
    eclass: "2"
    cost: 1
  Edge-3476130915693767771:
    op: Edge
    children:
      - i64-11743562013128004906
      - i64-5871781006564002453
      - i64-11743562013128004906
    eclass: "0"
    cost: 1
  Edge-14928484768540309164:
    op: Edge
    children:
      - i64-5040379952546458196
      - i64-17615343019692007359
      - i64-17615343019692007359
    eclass: "3"
    cost: 1
  Edge-12184509778494182535:
    op: Edge
    children:
      - i64-5040379952546458196
      - i64-5871781006564002453
      - i64-5871781006564002453
    eclass: "1"
    cost: 1
root_eclasses: []
class_data:
  i64-5535299385761720939:
    type: i64
  Unit-0:
    type: Unit
  i64-5871781006564002453:
    type: i64
  i64-5040379952546458196:
    type: i64
  i64-11743562013128004906:
    type: i64
  i64-17615343019692007359:
    type: i64
  "1":
    type: edge
  i64-0:
    type: i64
  "0":
    type: edge
  "3":
    type: edge
  "2":
    type: edge

